genetic_algorithm/qlearning.py

```diff
@@ -7,40 +7,36 @@
 import numpy as np
 import pickle
 
+GAMMA = 0.9
+LEARNING_RATE = 0.7
+
 with open("Qtable.txt", "rb") as f:
     """
     This function is responsible for loading Qtable.txt if already present
     """
     Q_table = pickle.load(f)
 
 
-def Action(state, Q_table):
+def next_best_action(state: int, Q_table: np.ndarray) -> int:
     """
-    This function returns the most suitable action value from the qtable.
+    Return the most suitable action value from Q_table or a random action
+    if there is no np.argmax(Q_table[state]).
+    >>> next_best_action(1, []) in [0, 1, 2, 3]
+    True
     """
     action = np.argmax(Q_table[state])
-    if action is None:
-        action = randomAction()
-    return action
-
-
-def randomAction():
-    """
-    This function returns random action just in case the Qtable values return None.
-    """
-    action = np.random.choice([0, 1, 2, 3])
-    return action
+    return action if action is not None else np.random.choice([0, 1, 2, 3])
 
 
-def StateActionReward(player, x_food, y_food):
+def state_action_reward(player, x_food, y_food):
```

Comment:
Type hints?

Reply:
Making changes. Thank you for your help!